src/translate/pddl_parser/parsing_functions.py

```diff
@@ -188,121 +379,271 @@ def add_effect(tmp_effect, result):
         return pddl.ConditionalEffect(condition, effect)
     elif tag == "increase":
-        assert len(alist) == 3
-        assert alist[1] == ['total-cost']
-        assignment = parse_assignment(alist)
+        if len(alist) != 3 or alist[1] != ["total-cost"]:
+            context.error("'increase' expects two arguments",
+                          alist, syntax=SYNTAX_EFFECT_INCREASE)
+        assignment = parse_assignment(context, alist)
         return pddl.CostEffect(assignment)
     else:
         # We pass in {} instead of type_dict here because types must
         # be static predicates, so cannot be the target of an effect.
-        return pddl.SimpleEffect(parse_literal(alist, {}, predicate_dict))
-
-
-def parse_expression(exp):
-    if isinstance(exp, list):
-        functionsymbol = exp[0]
-        return pddl.PrimitiveNumericExpression(functionsymbol, exp[1:])
-    elif exp.replace(".", "").isdigit():
-        return pddl.NumericConstant(float(exp))
-    elif exp[0] == "-":
-        raise ValueError("Negative numbers are not supported")
-    else:
-        return pddl.PrimitiveNumericExpression(exp, [])
-
-def parse_assignment(alist):
-    assert len(alist) == 3
-    op = alist[0]
-    head = parse_expression(alist[1])
-    exp = parse_expression(alist[2])
-    if op == "=":
-        return pddl.Assign(head, exp)
-    elif op == "increase":
-        return pddl.Increase(head, exp)
-    else:
-        assert False, "Assignment operator not supported."
-
-
-def parse_action(alist, type_dict, predicate_dict):
-    iterator = iter(alist)
-    action_tag = next(iterator)
-    assert action_tag == ":action"
-    name = next(iterator)
-    parameters_tag_opt = next(iterator)
-    if parameters_tag_opt == ":parameters":
-        parameters = parse_typed_list(next(iterator),
-                                      only_variables=True)
-        precondition_tag_opt = next(iterator)
-    else:
-        parameters = []
-        precondition_tag_opt = parameters_tag_opt
-    if precondition_tag_opt == ":precondition":
-        precondition_list = next(iterator)
-        if not precondition_list:
-            # Note that :precondition () is allowed in PDDL.
-            precondition = pddl.Conjunction([])
+        return pddl.SimpleEffect(parse_literal(context, alist, {}, predicate_dict))
+
+
+def parse_expression(context, exp):
+    with context.layer("Parsing expression"):
+        if isinstance(exp, list):
+            if len(exp) < 1:
+                context.error("Expression cannot be an empty block.",
+                              syntax=SYNTAX_EXPRESSION)
+            functionsymbol = exp[0]
+            return pddl.PrimitiveNumericExpression(functionsymbol, exp[1:])
+        elif exp.replace(".", "").isdigit() and exp.count(".") <= 1:
+            return pddl.NumericConstant(float(exp))
+        elif exp[0] == "-":
+            context.error("Expression cannot be a negative number",
```

Comment:
I liked the old phrasing "Negative numbers are not supported" better, mainly because it does not use the word "expression" and thus is still understandable if you're not too familiar with PDDL terminology.

Reply:
Sounds reasonable, I support changing it back.

Reply:
For what it's worth, I think we don't support fractional numbers either. I think it's a mistake that we accept ".".